udata/core/discussions/models.py

```diff
@@ -46,19 +69,47 @@ def spam_report_message(self, breadcrumb):
 
 class Discussion(SpamMixin, db.Document):
     user = db.ReferenceField("User")
+    organization = db.ReferenceField("Organization")
+
     subject = db.GenericReferenceField()
     title = db.StringField(required=True)
     discussion = db.ListField(db.EmbeddedDocumentField(Message))
     created = db.DateTimeField(default=datetime.utcnow, required=True)
     closed = db.DateTimeField()
     closed_by = db.ReferenceField("User")
+    closed_by_organization = db.ReferenceField("Organization")
     extras = db.ExtrasField()
 
     meta = {
         "indexes": ["user", "subject", "-created"],
         "ordering": ["-created"],
     }
 
+    @property
+    def permissions(self):
+        from udata.auth import Permission
+        from udata.core.discussions.permissions import CloseDiscussionPermission
+
+        return {
+            "delete": Permission(),
+            "edit": CloseDiscussionPermission(self),
```

Comment:
I'm a bit lost about `CloseDiscussionPermission` to `edit` a Discussion?

Reply:
I refactor the permissions in 55b0bb52c, tell me what you think :-)